modules/RTC/JitsiLocalTrack.js

```diff
@@ -23,22 +23,24 @@ const logger = getLogger(__filename);
  * @returns {Promise}
  */
 function createMuteUnmutePromise(track, mute) {
-    if (track.inMuteOrUnmuteProgress) {
-        return Promise.reject(
-            new JitsiTrackError(JitsiTrackErrors.TRACK_MUTE_UNMUTE_IN_PROGRESS)
-        );
+    // If there is a mute / unmute operation in progress, chain this operation
+    // so it runs afterwards. We do it both on `then` and `catch` because
+    // promises don't have a `finally` yet.
+    if (track._mutePromise) {
+        return track._mutePromise
+            .then(() => track._setMute(mute))
+            .catch(() => track._setMute(mute));
```

Comment:
1. I'm afraid I don't see that the above does "chain" fully. `inMuteOrUnmuteProgress` was making sure that one doesn't enter `JitsiLocalTrack._setMute` while a previous method call hasn't exited it. `_mutePromise` does not do that. If one invokes `JitsiLocalTrack.mute`/`unmute` 3 or more times quickly enough - A, B & C, then A will enter `_setMute`, B will wait for A to enter `_setMute`, but C will wait for A or nothing depending on whether A has been fulfilled/rejected by that time and in either of the two cases C will be allowed to enter `_setMute` while B is not guaranteed to have exited it.

2. Promise.then(onFulfilled).catch(onRejected) is not the same as Promise.then(onFulfilled, onRejected). And to me it looks like the two uses `createMuteUnmutePromise` rather want the second construct really.

Let's see what we have now:

`track._setMute` is fulfilled, then `track._mutePromise = undefined` throws so we catch that and do `track._mutePromise = undefined` again. Really? Anyway, `track._mutePromise = undefined` will probably not throw so we only get source code that's strange but should practically not happen.

But the problem in the new source code is more realistic: `track._,mutePromise` is fulfilled, so we do `track._setMute(mute)` which throws, but we catch that and we do `track._setMute(mute)` again. Was that really the intention of the source code?

3. I think the comment about `finally` should be removed. I don't really know what you mean by `finally` in the context of Promises but I guess you want something like `Promise.then(finally, finally)`.